packages/web/src/js/ScrollView.ts

```diff
@@ -0,0 +1,127 @@
+// The `ScrollView` component is currently untested. This is because it is not possible to test it in a headless
+// environment (tests fail on `getBoundingClientRect` that we need to mock somehow).
+
+import BaseComponent from './BaseComponent';
+import { EventHandler, SelectorEngine } from './dom';
+import { debounce, enableToggleAutoloader } from './utils';
+
+const NAME = 'scrollView';
+const DEBOUNCE_DELAY_MS = 50;
+const DEFAULT_DIRECTION = 'vertical';
+const VIEWPORT_SELECTOR = '[data-spirit-element="viewport"]';
+const CONTENT_SELECTOR = '[data-spirit-element="content"]';
+const IS_SCROLLED_AT_START_CLASS_NAME = 'is-scrolled-at-start';
+const IS_SCROLLED_AT_END_CLASS_NAME = 'is-scrolled-at-end';
+const RESIZE_EVENT = 'resize';
+const SCROLL_EVENT = 'scroll';
+
+// Method `getElementsPositionDifference` sometimes returns floating point values that results in inaccurate detection
+// of start/end. It is necessary to accept this inaccuracy and take every value less or equal to 1px as start/end.
+const EDGE_DETECTION_INACCURACY_PX = 1;
+
+export type Direction = 'horizontal' | 'vertical';
+
+interface ICurrentPosition {
+  [key: string]: number;
+  bottom: number;
+  left: number;
+  right: number;
+  top: number;
+}
+
+class ScrollView extends BaseComponent {
+  content: HTMLElement;
+  currentPosition: ICurrentPosition;
+  direction: Direction;
+  isScrolledAtEnd: boolean;
+  isScrolledAtStart: boolean;
+  scrollPositionEnd: string;
+  scrollPositionStart: string;
+  viewport: HTMLElement;
+
+  static get NAME(): string {
+    return NAME;
+  }
+
+  static get DATA_KEY(): string {
+    return `${this.NAME}`;
+  }
+
+  constructor(element: HTMLElement) {
+    super(element);
+
+    this.currentPosition = {
+      bottom: 0,
+      left: 0,
+      right: 0,
+      top: 0,
+    };
+    this.direction = this.element.dataset?.direction
+      ? (String(this.element.dataset.direction) as Direction)
+      : DEFAULT_DIRECTION;
+
+    this.content = SelectorEngine.findOne(CONTENT_SELECTOR, this.element) as HTMLElement;
```

Comment:
This solves the using of `SpiritElement`.
```suggestion
    this.content = SelectorEngine.findOne(CONTENT_SELECTOR, this.element);
```